pkg/fakeoperator/fakeoperator.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fakeoperator
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	zap2 "go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	corescheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	fakev1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/fake/v1alpha1"
+	"github.com/kubermatic/kubecarrier/pkg/fakeoperator/internal/controllers"
+)
+
+type flags struct {
+	metricsAddr          string
+	enableLeaderElection bool
+	verbosity            int8
+	healthAddr           string
+}
+
+func NewFakeOperator() *cobra.Command {
+	flags := flags{}
+
+	cmd := &cobra.Command{
+		Args:  cobra.NoArgs,
+		Use:   "e2e-operator",
+		Short: "e2e-operator runs the dummy joke operator for e2e testing purposes",
+	}
+
+	cmd.Flags().StringVar(&flags.metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	cmd.Flags().BoolVar(&flags.enableLeaderElection, "enable-leader-election", false,
+		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
+	cmd.Flags().StringVar(&flags.healthAddr, "health-addr", ":9440", "The address the health endpoint binds to.")
+	cmd.Flags().Int8VarP(&flags.verbosity, "verbosity", "v", 4, "log level version")
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		return run(flags, ctrl.Log.WithName("setup"))
+	}
+	return cmd
+}
+
+func run(flags flags, log logr.Logger) error {
+	var (
+		scheme = runtime.NewScheme()
+	)
+
+	_ = fakev1alpha1.AddToScheme(scheme)
```

Comment:
check those errors, you could even panic via `utilruntime.Must`  regarding the scheme, so far we placed it as package-private variable and initialized it in the `init` function. Might do the same here.

Reply:
I did this for webhooks PR already. backported here